### ISSUE TITLE
Issues with master under linux after recent merges

### ISSUE THREAD

**Issue by LynxAbraxas** [civctp2/civctp2#66]
In relation to #63, while playing up to 3450BC (22 turns) on linux from DI, I had about 6 dead-locks of the game (compiled at https://github.com/civctp2/civctp2/commit/9371b6c1094c65c40493cd4f8f86e9c5b08ec589), which is far, far more often than I had crashes (no dead-locks) before (one in about 3 hours).
Also the OGG playing (that used to work with https://github.com/civctp2/civctp2/commit/74c4f072fa185716eef73fd0393689a5112a27ad) is broken, game sounds and video (with audio) are functional though.
@MartinGuehmann any idea what commits could be the source for these changes?

**Comment by MartinGuehmann** [comment]
The dead-locks are resolved in PR #61. Now it is crashing again. I merged them quite quickly, when I realized that you and others shouldn't test these things. I guess these PRs are a bit special.

For the OGG playing, I have no idea right now, for that I would use TortoiseGit, which is not available on my Kubuntu. I got a debug version compiling with some of the debug tools. However for now it trows some Asserts that aren't thrown on Windows and then crashes. However, compiling a final version seems to be fine. Probably, I have now enough for a WIP pull request.

Obviously it is something with sound, I edited the sound manager, but probably there are some other places to look for.

**Comment by LynxAbraxas** [comment]
> The dead-locks are resolved in PR #61. Now it is crashing again. 

Sadly not, the game at https://github.com/civctp2/civctp2/commit/9371b6c1094c65c40493cd4f8f86e9c5b08ec589 already includes #61. I think #61 has not reverted all changes. I'll take a look.
Odd though that the dead-locks appear much, much more often than the former crashes.

**Comment by MartinGuehmann** [comment]
I looked it again, looks like I misunderstood what TortoiseGit was showing me, so that I just undid one of your edits on sdl_blitter.cpp:

```
        // lock both surfaces
        // note that there is a chance of a deadlock if a different
        // thread locks the two surfaces in the opposite order
        /*SDL_LockMutex(sdlDest->m_bltMutex); //http://www.libsdl.org/cgi/docwiki.cgi/SDL_5fBlitSurface says don't lock surface! mutex lock OK but seems not needed
        SDL_LockMutex(sdlSrc->m_bltMutex);*/
```

Looks like the dead locks are a well known problem.

Well, at least I should now have got it, it is PR #67.

**Comment by MartinGuehmann** [comment]
The OGGs, I have checked the sound manger, but that doesn't seem to be the place. I outcommented some OGG related code, but strangely it was only used for Windows.

**Comment by MartinGuehmann** [comment]
The other guess, I have is c3files.cpp. I merged some code in for CD recognition and also debugged, so that the game finds the CD. Another place could be tracklen.cpp, that deals also with music, I think I also merged in Linux branch code.

What I can say is that not only the OGGs do not play, but also the music from the CD does not play. However, I cannot say whether it is the same problem or those two issues are unrelated.

**Comment by LynxAbraxas** [comment]
@ptitSeb Do you mind having a look what code or yours got lost that made OGG palying possible?

**Comment by ptitSeb** [comment]
I'll try to have a look.

**Comment by LynxAbraxas** [comment]
That's great! Thanks.

**Comment by MartinGuehmann** [comment]
Actually, were do go the oggs, I ripped the music from the CD. When I asked on Apolyton it was:

../ctp2_program/ctp/Sounds/ for the GoG version.

**Comment by MartinGuehmann** [comment]
The place is supposed to be ./music/Track02.ogg, relative to the path of the working directory, most likely were the ctp2 executable is.

My oggs are called:

02 - Carey Chico - A Fortiori.ogg
03 - Carey Chico - Pacari Tampu.ogg
04 - Carey Chico - Amaethon.ogg
05 - Carey Chico - A 'Tugewinu.ogg
06 - Carey Chico - Summum Bonum.ogg
07 - Carey Chico - Casus Belli.ogg
08 - Carey Chico - Ankh Djet.ogg
09 - Carey Chico - Shaolin San.ogg
10 - Michael Harriton - Quiet Outbreak.ogg
11 - Michael Harriton - Kali Dance.ogg

**Comment by LynxAbraxas** [comment]
With the docker image of ctp2DF, I bind mount the music to /opt/ctp2/ctp2_program/ctp/music/, as described here:
https://github.com/civctp2/civctp2/commit/9e9185e8887a6d5636a16b3f41a98d80b0d4dc03#diff-04c6e90faac2675aa89e2176d2eec7d8
They have to be named (or linked) to filnames of the pattern `Track%02d.ogg`:
https://github.com/civctp2/civctp2/blob/3e6e7134c6179d54c6a043484cfffa176811dfe9/ctp2_code/sound/soundmanager.cpp#L920
and it starts from Track02.ogg:
https://github.com/civctp2/civctp2/blob/3e6e7134c6179d54c6a043484cfffa176811dfe9/ctp2_code/sound/soundmanager.cpp#L217

**Comment by MartinGuehmann** [comment]
That's what I figured, I changed it to 

`if(CI_FileExists("./music/Track02.ogg")) { `

And I got them playing, I just have some other changes in that might influence it.

**Comment by LynxAbraxas** [comment]
That's great, I wonder though why `./` makes a difference.

**Comment by MartinGuehmann** [comment]
I don't know whether this makes a difference, since I have some other changes in, I am just sorting them and add at least one pull request.

**Comment by MartinGuehmann** [comment]
It could also be #78. At least that explains why removing include files or changing their order matters, it could cause that some SDL specefic code was not included, even so it was supposed so.

**Comment by MartinGuehmann** [comment]
So it works also without #78. I just haven't tried without ./

**Comment by LynxAbraxas** [comment]
Sadly, for me not. Tested with https://github.com/civctp2/civctp2/commit/3e6e7134c6179d54c6a043484cfffa176811dfe9 and https://github.com/LynxAbraxas/civctp2/commit/99ead4b7a99d91144476ea24c2d3f85735c3c5e4. Any idea what could make the difference? OGG-playing works for me upto https://github.com/civctp2/civctp2/commit/74c4f072fa185716eef73fd0393689a5112a27ad.

**Comment by LynxAbraxas** [comment]
Looking at the changes
https://github.com/civctp2/civctp2/commits/master/ctp2_code/sound/soundmanager.cpp introduced since https://github.com/civctp2/civctp2/commit/74c4f072fa185716eef73fd0393689a5112a27ad with `gitk -- ctp2_code/sound/soundmanager.cpp` (color words, ignore space change) I have my doupts that any changes to `ctp2_code/sound/soundmanager.cpp` cause the problems. @MartinGuehmann did you change anything else concerning playing sound with SDL/linux since https://github.com/civctp2/civctp2/commit/74c4f072fa185716eef73fd0393689a5112a27ad?

**Comment by MartinGuehmann** [comment]
Now, it is really confusing. Suddenly it plays the audio from the CD and also points out if no CD is inside. And when I quit it tells me "Detected Ogg Music track, using that instead of CD Audio".

The only thing, I did was recompiling and unplugging the hard drive that made trouble.

**Comment by LynxAbraxas** [comment]
Ah, I know those strange cases. The only thing that really helped me to get rid of those is to use docker environments;-) They can put your teeth on edge too, though.

**Comment by MartinGuehmann** [comment]
Afterwards, I kicked out the failing drive, I just updated and installed Ddrescue and Ddrescue-GUI, which I couldn't install with the failing drive, even so the partitions weren't mounted it still was doing something with it.

Apart from that I also added some code from the Linux branch, which is supposed to recognize the CD and find the tracks on the CD. However, it just didn't do it, until now. Linux updates, but if you are in a Dockker environment then you should be fine.

**Comment by LynxAbraxas** [comment]
This OGG problem is really odd. I tested if it works when reverting the changes concerning 
`soundmanager.*` https://github.com/LynxAbraxas/civctp2/commits/debugOGGplaying
but even with https://github.com/LynxAbraxas/civctp2/commit/1edbb0d36603e08766d8cecc983990bf9180b239 (reverting back to @ptitSeb laste state) OGGs do not play for me.
So I would conclude that the problems come from some other change than those to `soundmanager.*` @ptitSeb @MartinGuehmann any idea what could be wrong?

**Comment by LynxAbraxas** [comment]
Solved with #83.
